PWGJE/Tasks/gammajettreeproducer.cxx

```diff
@@ -0,0 +1,328 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "Framework/ASoA.h"
+#include "Framework/AnalysisDataModel.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/HistogramRegistry.h"
+#include "CCDB/BasicCCDBManager.h"
+
+#include "Common/Core/RecoDecay.h"
+#include "Common/Core/TrackSelection.h"
+#include "Common/Core/TrackSelectionDefaults.h"
+#include "Common/DataModel/EventSelection.h"
+#include "Common/DataModel/TrackSelectionTables.h"
+
+#include "PWGJE/Core/FastJetUtilities.h"
+#include "PWGJE/Core/JetDerivedDataUtilities.h"
+#include "PWGJE/DataModel/Jet.h"
+#include "PWGJE/DataModel/GammaJetAnalysisTree.h"
+
+#include "EMCALBase/Geometry.h"
+#include "EMCALCalib/BadChannelMap.h"
+#include "PWGJE/DataModel/EMCALClusters.h"
+#include "DataFormatsEMCAL/Cell.h"
+#include "DataFormatsEMCAL/Constants.h"
+#include "DataFormatsEMCAL/AnalysisCluster.h"
+#include "TVector2.h"
+#include "EventFiltering/Zorro.h"
+
+#include "CommonDataFormat/InteractionRecord.h"
+
+#include "EventFiltering/filterTables.h"
+
+// \struct GammaJetTreeProducer
+/// \brief Task to produce a tree for gamma-jet analysis, including photons (and information of isolation) and charged and full jets
+/// \author Florian Jonas <florian.jonas@cern.ch>, UC Berkeley/LBNL
+/// \since 02.08.2024
+///
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+using bcEvSelIt = o2::soa::Join<o2::aod::BCs, o2::aod::BcSels>::iterator;
+using selectedClusters = o2::soa::Filtered<o2::aod::JClusters>;
+
+#include "Framework/runDataProcessing.h"
+
+struct GammaJetTreeProducer {
+  // analysis tree
+  // charged jets
+  // photon candidates
+  Produces<aod::GjChargedJets> chargedJetsTable;
+  Produces<aod::GjEvents> eventsTable;
+  Produces<aod::GjGammas> gammasTable;
+
+  HistogramRegistry mHistograms{"GammaJetTreeProducerHisto"};
+
+  // configure zorro
+  Service<o2::ccdb::BasicCCDBManager> ccdb;
+  Zorro zorro;
```

Comment:
here you should not use zorro anymore and get the decisions directly from the triggerSel coloum in JCollisions. You can find examples in jetFinderQA